contrib/colab/mnist-training.ipynb

```diff
@@ -205,28 +207,25 @@
         "def train_mnist():\n",
         "  torch.manual_seed(1)\n",
         "  \n",
-        "  if not xm.is_master_ordinal():\n",
-        "    # Barrier: Wait until master is done downloading\n",
-        "    xm.rendezvous('download_only_once')\n",
-        "\n",
-        "  # Get and shard dataset into dataloaders\n",
-        "  norm = transforms.Normalize((0.1307,), (0.3081,))\n",
-        "  train_dataset = datasets.MNIST(\n",
-        "      FLAGS['datadir'],\n",
-        "      train=True,\n",
-        "      download=True,\n",
-        "      transform=transforms.Compose(\n",
-        "          [transforms.ToTensor(), norm]))\n",
-        "  test_dataset = datasets.MNIST(\n",
-        "      FLAGS['datadir'],\n",
-        "      train=False,\n",
-        "      download=True,\n",
-        "      transform=transforms.Compose(\n",
-        "          [transforms.ToTensor(), norm]))\n",
+        "  def get_dataset():\n",
+        "    norm = transforms.Normalize((0.1307,), (0.3081,))\n",
+        "    train_dataset = datasets.MNIST(\n",
+        "        FLAGS['datadir'],\n",
+        "        train=True,\n",
+        "        download=True,\n",
+        "        transform=transforms.Compose(\n",
+        "            [transforms.ToTensor(), norm]))\n",
+        "    test_dataset = datasets.MNIST(\n",
+        "        FLAGS['datadir'],\n",
+        "        train=False,\n",
+        "        download=True,\n",
+        "        transform=transforms.Compose(\n",
+        "            [transforms.ToTensor(), norm]))\n",
+        "    \n",
+        "    return train_dataset, test_dataset\n",
         "  \n",
-        "  if xm.is_master_ordinal():\n",
-        "    # Barrier: Master done downloading, other workers can proceed\n",
-        "    xm.rendezvous('download_only_once')\n",
+        "  # Only first run of get_dataset downloads dataset locally\n",
```

Comment:
I would explain a bit better.
Like:

Using the serial executor avoids multiple processes to download the same data.

Reply:
Done.